std/bitmanip.d

```diff
@@ -37,18 +37,12 @@ version(unittest)
 }
 
 
-private string myToStringx(ulong n)
-{
-    enum s = "0123456789";
-    if (n < 10)
-        return s[cast(size_t)n..cast(size_t)n+1];
-    else
-        return myToStringx(n / 10) ~ myToStringx(n % 10);
-}
-
 private string myToString(ulong n)
 {
-    return myToStringx(n) ~ (n > uint.max ? "UL" : "U");
+    import core.internal.string;
+    UnsignedStringBuf buf;
+    auto s = unsignedToTempString(n, buf);
+    return cast(string)s ~ (n > uint.max ? "UL" : "U");
```

Comment:
Looking at core.internal.string, it looks like `UnsignedStringBuf` is a static array of `char`, and `unsignedToTempString` returns a slice of its argument, so doesn't this return a slice of a static array that's a local variable? It looks like `s` is going to need to do be `idup`ed.